### PR TITLE
feat: implement autoapi

### LINKED ISSUE #759: Set up `ansys-sphinx-theme[autoapi]` for API reference guide
@dipinknair I think we should implement sphinx-autoapi instead using `ansys-sphinx-theme[autoapi]==0.16.5`. Using autoapi should generate an API reference for all files in the src directory. I can help implement it if you want

_Originally posted by @klmcadams in https://github.com/ansys/pymechanical/issues/758#issuecomment-2153272305_

### PR DESCRIPTION
**Pull request by klmcadams**
Implement ansys-sphinx-theme[autoapi] and add docstrings to constants. Please feel free to change the docstrings if there are better descriptions. I deleted the `doc/source/api` folder to remove the manual API information, but if you think we should keep some of these files lmk

Closes #759

### CODE REVIEW

File: doc/source/index.rst [hunk 1]
```diff
@@ -47,6 +47,7 @@
    user_guide_session/index
    user_guide_embedding/index
    user_guide_scripting/index
+   architecture
```
Comment (dipinknair) [inline doc/source/index.rst:50]: @koubaa pinging to see whether this should be there in index.

**Review by RobPasMue — APPROVED** [review]

File: doc/source/ansys_tools_path_api/path.rst [hunk 2]
```diff
@@ -15,5 +15,4 @@ of the installation path of the Mechanical application.
    find_mechanical
    get_mechanical_path
    change_default_mechanical_path
-   save_mechanical_path
-
+   save_mechanical_path
```
Comment (klmcadams) [inline doc/source/ansys_tools_path_api/path.rst:18]: @koubaa Where should this file be referenced in the docs? api/index.rst?

Comment (koubaa) [reply doc/source/ansys_tools_path_api/path.rst:18]: I suggest api/path.rst which is linked to by api/index.rst

File: doc/source/user_guide_session/index.rst [hunk 3]
```diff
@@ -21,14 +21,13 @@ to a remote Mechanical session.
 Overview
 --------
 
-The :func:`launch_mechanical() <ansys.mechanical.core.launch_mechanical>` method
-creates an instance of the :class:`Mechanical <ansys.mechanical.core.mechanical.Mechanical>`
+The `launch_mechanical() <https://mechanical.docs.pyansys.com/version/stable/api/ansys/mechanical/core/mechanical/index.html#mechanical.launch_mechanical>`_ method
```
Comment (dipinknair) [inline doc/source/user_guide_session/index.rst:24]: What happens if we have some change in dev version of launch.mechanical ? Will it be redirected to stable version in dev docs ?

Comment (klmcadams) [reply doc/source/user_guide_session/index.rst:24]: Yes I believe so

Comment (klmcadams) [reply doc/source/user_guide_session/index.rst:24]: Should we change the links to version/dev then? but then we'll have that issue with the stable docs

**Review by koubaa — APPROVED** [review]

**Review by dipinknair — APPROVED** [review]
Awesome !

**Pull request merged**